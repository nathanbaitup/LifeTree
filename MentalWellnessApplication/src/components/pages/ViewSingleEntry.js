import React, { useState, useEffect } from 'react';
import { Text, View, StyleSheet, ImageBackground, TextInput, Image, TouchableOpacity } from 'react-native';

import firestore from '@react-native-firebase/firestore';

export default function ViewSingleEntry(props) {
    
    const [allEntries, setAllEntries] = useState([]);
    const [journalEntry, setJournalEntry] = useState([]);

    
    const userID = props.extraData;
    const onBack = props.onBack;
    const entryID = props.currentEntryID;

    const journalsRef = firestore().collection('journalList');
    
    // Selects all journal entries where the user ID matches the authorID and sorts the list by newest date first.
    useEffect(() => {
        journalsRef
            .where('authorID', '==', userID)
            .orderBy('createdAt', 'desc')
            .onSnapshot(
                querySnapshot => {
                    const newJournals = [];
                    querySnapshot.forEach((doc) => {
                        const journal = doc.data();
                        journal.id = doc.id;
                        newJournals.push(journal);
                    });
                    setAllEntries(newJournals);
                },
                error => {
                    alert(error);
                }
            );
            let selectedEntry = allEntries.find(entry => entry.createdAt === entryID);
            setJournalEntry(selectedEntry); 
    }, []);




        return (
            // TODO: Import updated add an entry page from entries branch to get the mood images to highlight the users mood for an entry.
            <ImageBackground source={require('../resources/img/background.png')} style={{ width: '100%', height: '100%', opacity: 50 }} >
                <View style={styles.contentContainer}>
                    <Text style={styles.dateTitle}>Date of Entry:  {console.log(allEntries)} </Text>
                    <Text style={styles.header}>How you were feeling: </Text>

                    <View style={styles.moodModules}>
                        <View style={journalEntry.moodSelected === 'Angry' ? styles.moodModSelected : styles.moodModUnselected} >
                            <Image source={require('../resources/img/faces/angry.png')} style={styles.moodFaces} />
                            <Text>Angry</Text>
                        </View>
                        <View style={journalEntry.moodSelected === 'Sad' ? styles.moodModSelected : styles.moodModUnselected}>
                            <Image source={require('../resources/img/faces/sad.png')} style={styles.moodFaces} />
                            <Text>Sad</Text>
                        </View>
                        <View style={journalEntry.moodSelected === 'Meh' ? styles.moodModSelected : styles.moodModUnselected}>
                            <Image source={require('../resources/img/faces/meh.png')} style={styles.moodFaces} />
                            <Text>Meh</Text>
                        </View>
                        <View style={journalEntry.moodSelected === 'Happy' ? styles.moodModSelected : styles.moodModUnselected}>
                            <Image source={require('../resources/img/faces/happy.png')} style={styles.moodFaces} />
                            <Text>Happy</Text>
                        </View>
                    </View>

                    <Text style={[{ textAlign: 'center' }, {color: journalEntry.moodSelected === 'Happy' ? '#108206' : journalEntry.moodSelected === 'Meh' ? '#e38e07' : journalEntry.moodSelected === 'Sad' ? '#112dec' : '#f90505' }]}>
                       You were feeling: {journalEntry.moodSelected}
                    </Text>
                    <View>
                        <Text style={styles.journalHeader}>Your comments for the day: </Text>
                        <TextInput style={styles.journalEntry}
                            placeholder='Feel free to dump as much or as little information in here as you want. We wont judge you at all. We promise! '
                            numberOfLines={5}
                            multiline
                            editable={false}
                            value={journalEntry.journalText}
                        />
                        <View style={styles.returnButtonContainer}>
                            <TouchableOpacity style={styles.returnButton} onPress={onBack}>
                                <Text style={styles.returnText}>Return to Entries</Text>
                            </TouchableOpacity>
                        </View>
                    </View>
                </View>
            </ImageBackground>
        );
    }


// The styling for the page.
const styles = StyleSheet.create({
    contentContainer: {
        margin: 20,
        padding: 10,
        backgroundColor: '#FFFFFF',
        borderRadius: 25,
    },
    header: {
        fontSize: 18,
        fontWeight: 'bold',
        color: '#000000',
        padding: 10,
    },
    moodModules: {
        flexDirection: 'row',
        paddingBottom: 50,
    },
    moodModUnselected: {
        marginTop: 10,
        marginLeft: 5,
        textAlign: 'center',
        paddingLeft: 10,
    },
    moodModSelected: {
        marginTop: 10,
        marginLeft: 5,
        textAlign: 'center',
        borderWidth: 1,
        borderColor: '#000000',
        borderRadius: 5,
        paddingLeft: 10,  
    },
    moodFaces:{
        marginTop: 10,
        height: 50,
        width: 66,
        alignItems: 'center',
    },
    journalEntry: {
        marginTop: 5,
        marginBottom: 20,
        backgroundColor: '#ebebeb',
        borderRadius: 10,
        maxHeight: 250,
        height: 150,
    },
    journalHeader: {
        fontSize: 18,
        fontStyle: 'italic',
        fontWeight: 'bold',
        color: '#000000',
        padding: 10,
    },
    returnButtonContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
    },
    returnButton: {
        padding: 15,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'orange',
        borderRadius: 15,
        width: 150,
    },
    returnText: {
        textAlign: 'center',
        fontWeight: 'bold',
        fontSize: 18,
        color: '#000000',
    },
    dateTitle: {
        padding: 15,
        fontWeight: 'bold',
        fontSize: 18,
    }
});
