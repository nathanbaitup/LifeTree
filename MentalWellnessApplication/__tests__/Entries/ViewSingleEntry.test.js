import 'react-native';
import React from 'react';
import ViewSingleEntry from '../../src/components/pages/Entries/ViewSingleEntry';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('Checks to see if the View Single Entry Page renders', () => {
  const tree = renderer.create(<ViewSingleEntry />).toJSON();
  expect(tree).toMatchSnapshot();
});