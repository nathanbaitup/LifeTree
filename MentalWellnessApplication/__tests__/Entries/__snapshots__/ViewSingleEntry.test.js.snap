// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks to see if the View Single Entry Page renders 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "opacity": 50,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/resources/img/background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 25,
        "margin": 20,
        "padding": 10,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 18,
              "fontWeight": "bold",
              "padding": 15,
            }
          }
        >
          Date of Entry: 
          
           
        </Text>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
        >
          How you were feeling: 
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingBottom": 50,
            }
          }
        >
          <View
            style={
              Object {
                "marginLeft": 5,
                "marginTop": 10,
                "paddingLeft": 10,
                "textAlign": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/resources/img/faces/angry.png",
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "marginTop": 10,
                  "width": 66,
                }
              }
            />
            <Text>
              Angry
            </Text>
          </View>
          <View
            style={
              Object {
                "marginLeft": 5,
                "marginTop": 10,
                "paddingLeft": 10,
                "textAlign": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/resources/img/faces/sad.png",
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "marginTop": 10,
                  "width": 66,
                }
              }
            />
            <Text>
              Sad
            </Text>
          </View>
          <View
            style={
              Object {
                "marginLeft": 5,
                "marginTop": 10,
                "paddingLeft": 10,
                "textAlign": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/resources/img/faces/meh.png",
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "marginTop": 10,
                  "width": 66,
                }
              }
            />
            <Text>
              Meh
            </Text>
          </View>
          <View
            style={
              Object {
                "marginLeft": 5,
                "marginTop": 10,
                "paddingLeft": 10,
                "textAlign": "center",
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/resources/img/faces/happy.png",
                }
              }
              style={
                Object {
                  "alignItems": "center",
                  "height": 50,
                  "marginTop": 10,
                  "width": 66,
                }
              }
            />
            <Text>
              Happy
            </Text>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "textAlign": "center",
              },
              Object {
                "color": "#f90505",
              },
            ]
          }
        >
          You were feeling: 
          
        </Text>
        <View>
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 16,
                "fontStyle": "italic",
                "fontWeight": "bold",
              }
            }
          >
            Your comments for the day: 
          </Text>
          <TextInput
            editable={false}
            multiline={true}
            numberOfLines={5}
            placeholder="Feel free to dump as much or as little information in here as you want. We wont judge you at all. We promise! "
            style={
              Object {
                "backgroundColor": "#ebebeb",
                "borderRadius": 10,
                "height": 150,
                "marginBottom": 20,
                "marginTop": 5,
                "maxHeight": 250,
              }
            }
            value=""
          />
          <Text
            style={
              Object {
                "color": "#000000",
                "fontSize": 16,
                "fontStyle": "italic",
                "fontWeight": "bold",
              }
            }
          >
            Obsession of the day?
          </Text>
          <TextInput
            multiline={true}
            numberOfLines={1}
            placeholder="Enter your random obsession here!"
            style={
              Object {
                "backgroundColor": "#ebebeb",
                "borderRadius": 10,
                "height": 40,
                "marginBottom": 20,
                "marginTop": 5,
                "maxHeight": 40,
              }
            }
            value=""
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "textAlign": "center",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={false}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "orange",
                  "borderRadius": 15,
                  "justifyContent": "center",
                  "opacity": 1,
                  "padding": 15,
                  "width": 150,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "textAlign": "center",
                  }
                }
              >
                Return to Entries
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
